### PR TITLE
Fix a lifetime issue in FoundationValueTypes.cpp

### PR DESCRIPTION
**Pull request by fredriss**
m_synth_up was keeping a reference to the locally created ObjCRuntimeSyntheticProvider.
Fix this by making the lifetime of ObjCRuntimeSyntheticProvider the same as the
lifetime of the frontend.

rdar://problem/43799990

### CODE REVIEW

**Comment by fredriss** [comment]
@swift-ci test and merge

File: source/Plugins/Language/Swift/FoundationValueTypes.cpp [hunk 1]
```diff
@@ -282,7 +284,8 @@ class URLComponentsSyntheticChildrenFrontEnd
     : public SyntheticChildrenFrontEnd {
 public:
   URLComponentsSyntheticChildrenFrontEnd(lldb::ValueObjectSP valobj_sp)
-      : SyntheticChildrenFrontEnd(*valobj_sp), m_synth_up(nullptr)
+      : SyntheticChildrenFrontEnd(*valobj_sp), m_synth_frontend_up(),
+        m_synth_backend_up()
```
Comment (adrian-prantl) [inline source/Plugins/Language/Swift/FoundationValueTypes.cpp:288]: Not really part of this patch, but: I suppose these should be named `_sp` (smartpointer) instead of `_up`?
